syntax = "proto3";
package user_proto;

message UserId{
    string id = 1;
}

message UserInfo {
    string id = 1;
    string fname = 2;
    string lname = 3;
    string email = 4;
    string ccode = 5;
    string phone_number = 6;
    string username = 7;
    string image = 8;
    string utype = 9;
    string organisation = 10;
}

message UserPermissionCheck{
    string id = 1;
    string path = 2;
    string method = 3;
}

message BooleanData{
    bool verified = 1;
}

service UserData {
    rpc GetUser(UserId) returns (UserInfo) {}
    rpc CheckUserPermission(UserPermissionCheck) returns (BooleanData) {}
}